cmd/delete.go

```diff
@@ -36,7 +37,28 @@ No local files are deleted.
 		PreRunE:           bindEnv("path", "confirm", "all", "namespace", "verbose"),
 		SilenceUsage:      true, // no usage dump on error
 		RunE: func(cmd *cobra.Command, args []string) error {
-			return runDelete(cmd, args, newClient)
+			// Layer 2: Catch technical errors and provide CLI-specific user-friendly messages
+			err := runDelete(cmd, args, newClient)
+			if err != nil && errors.Is(err, fn.ErrNameRequiredForDelete) {
```

Comment:
This is good!  I would just use a general "ErrNameRequired".  The fact it's "for delete" is implied by its context here (so you can reuse ErrNameRequired in other places where... a name is required)